File: src/Plotly.NET/Surface.fs
namespace Plotly.NET

open DynamicObj
open System

type Surface () =
    inherit DynamicObj () 

    static member init 
        (
            ?Count: int,
            ?Fill: float,
            ?Pattern: StyleParam.SurfacePattern,
            ?Show: bool
        ) =
            Surface()
            |> Surface.style
                (
                    ?Count      = Count,
                    ?Fill       = Fill,
                    ?Pattern    = Pattern,
                    ?Show       = Show
                )

    static member style
        (
            ?Count: int,
            ?Fill: float,
            ?Pattern: StyleParam.SurfacePattern,
            ?Show: bool
        ) =

            fun (surface: Surface) ->
                
                Count       |> DynObj.setValueOpt surface "count"
                Fill        |> DynObj.setValueOpt surface "fill"
                Pattern     |> DynObj.setValueOptBy surface "pattern" StyleParam.SurfacePattern.convert
                Show        |> DynObj.setValueOpt surface "show"

                surface

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Surface.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.Surface InitSurface
        (
            Optional<int> Count = default,
            Optional<double> Fill = default,
            Optional<StyleParam.SurfacePattern> Pattern = default,
            Optional<bool> Show = default
        ) 
            =>
                Plotly.NET.TraceObjects.Surface.init(
                    Count: Count.ToOption(),
                    Fill: Fill.ToOption(),
                    Pattern: Pattern.ToOption(),
                    Show: Show.ToOption()
                );
}

